### DISCUSSION TITLE
SBFspot

### DISCUSSION THREAD

**Discussion by shieling** [SBFspot/SBFspot#581] [Q&A] [answered]
I have recently installed SBFspot on my Dell Ubuntu (version 22.01) laptop. After setting up all the necessary applications I can now connect to my two SMA inverters. When I run SBFspot in the verbose mode the output seems to be working OK - it sees the inverters and shows the current KWh correctly for both inverters, but at the end I get the message that 'There is not data to export' and the resulting CSV file shows 0 to each day on the month and also for the current day. I have checked using both Sunny Beam and Sunny Explorer (on my Windows Computer) and all data in available from the inverters.
I have attached a file showing the output of my command line - SBFspot.3 -v -nosql
I hope someone can suggest what is going wrong - perhaps I have some configuration errors or maybe a bug in the system.
[output.pdf](https://github.com/SBFspot/SBFspot/files/9844153/output.pdf)

**Comment by sbf-** [comment]
I'm afraid it's a bug as it (SBFspot) treats your BT Repeater as an inverter:
```
SUSyID: 102 - SN: 170004481
Device Name: BT Repeater
Device Class: Communication products
Device Type: Bluetooth Repeater
Software Version: 12.12.104.R

SUSyID: 102 - SN: 170004481
DC Spot Data:
MPPT 1 Pdc: 0.000kW - Udc: 0.00V - Idc: 0.000A
MPPT 2 Pdc: 0.000kW - Udc: 0.00V - Idc: 0.000A
Calculated Total Pdc: 0.000kW
```

Can you provide a full debug log? (-d5 -v5)
Thanks

**Comment by shieling** [comment]
Thanks for responding. I have attached the full debug output.
[output.txt](https://github.com/SBFspot/SBFspot/files/9844732/output.txt)

**Comment by shieling** [comment]
I have tried again, but this time setting the .cfg file to read just the one inverter. This seems to output the data in the csv file for that inverter. Does this mean that I will have to run the SBFspot twice, one for each inverter, or can I combine the two in one run of the program?

**Reply by sbf-** [reply]
Yes, you can work around with two config files for each inverter and run SBFspot twice with -cfg option

**Comment by shieling** [comment]
Its a shame that I can't run the two inverters from the same config file and somehow ignore the repeater. However, I only need the csv files - as I use these to upload the data to my own website. I have previously used Sunny Explorer on windows, but now have gone away from windows. Would I be able to run the two instances at the same time, or would it be better (essential) to separate the run times. As the data is only available every 5 minutes I should be able to synchronise the two sets of data. I use my own database to populate from the csv files, so I don't need either the sqllite database or the export to PVoutput.

**Reply by sbf-** [reply]
> Would I be able to run the two instances at the same time

No, the first process will block the local BT adapter until finished

**Comment by shieling** [comment]
Thank you SBF for your responses. I have one more question before I close this thread...
As I do not require the SQLite database, if I make the SBFspot file, but do not make the SQLite file, will the SBFspot still operate if I make sure I use the correct aparameters (i.e. -nsql)?
I suppose the question is really 'will SBFspot work without SQlite installed if I use the correct parameters'?

**Reply by sbf-** [reply]
> will SBFspot work without SQlite installed if I use the correct parameters

It's as easy as building SBFspot with `make nosql`

**Comment by shieling** [comment]
Thanks very much for the information. Looks good to go now. I just need to work out the code to load my database with the data from the generated csv files. 
All seems to be working OK on my Ubuntu laptop.
My next task will be to build SBFspot on my MAC Studio  (MacOS M1 arm64)!!!! But that's another issue, unless you know someone who has already done this. Perhaps I'll open another thread for this.

**Comment by shieling** [comment]
PS. If I succeed in porting to the MAC Studio, I'll be donating something to the project, which I think is fantastic.

**Comment by Schielegerhard** [comment, marked as answer]
Adding my experience here for others struggling with SBFspot + 
BT Repeater (BTREP-IN.GR2) setups.

I had massive instability with the Repeater added to my system: 
only about 20% of the regular 5-minute runs succeeded, with various 
internal error codes and lots of mesh-handling problems. SBFspot 
sometimes saw the Repeater, sometimes not, sometimes picked up a 
neighbour's inverter on a different NetID instead.

Setup:
- 2× SB 3000TLST-21 (2012)
- 1× SMA Bluetooth Repeater BTREP-IN.GR2 (NetID 6)
- Windows 11, SBFspot 3.9.12 SQLite
- MIS_Enabled=1, NetID 6 throughout

The fix that worked for me: in SBFspot.cfg, change BTAddress from 
the inverter's BT MAC to the *Repeater's* BT MAC.

Before:
  BTAddress=00:80:25:XX:XX:XX  (inverter address)

After:
  BTAddress=00:80:25:YY:YY:YY  (Repeater address)

With this change, SBFspot connects directly to the Repeater first 
(single, stable point of contact) and then queries both inverters 
through the established mesh. Connection succeeds on first attempt, 
runs complete in ~10 seconds instead of 30-60s, and stability is now 
essentially 100% (LastTaskResult = 0 consistently).

The Repeater's BT MAC can be found by running SBFspot.exe -v -finq 
and looking for "Device X: 00:80:25:... -> Inverter" entries, or by 
running SBFspot.exe -scan when only the Repeater is in range.

This contrasts with the older suggestion of running two separate 
configs (one per inverter); a single config still works fine, as 
long as BTAddress points to the Repeater rather than an inverter.

Hope this helps others.
